### PR TITLE
[Instrumentation.OWIN] Bugfix - make Meter and Histogram as a singletons

### LINKED ISSUE #1654: The Owin metrics middleware creates new Meter instances for each request, causing massive lock contention
# Issue with OpenTelemetry.Instrumentation.Owin

* [ ] Feature Request
* [x] Bug

From src/OpenTelemetry.Instrumentation.Owin/Implementation/OwinInstrumentationMetrics.cs

```
    public static Meter Instance => new(MeterName, Assembly.GetPackageVersion());

    public static Histogram<double> HttpServerDuration => Instance.CreateHistogram<double>("http.server.request.duration", "s", "Duration of HTTP server requests.");
```
This creates a new Meter instance and a new Histogram every time OwinInstrumentationMetrics.HttpServerDuration is accessed. Creating new Meters is expensive and wrong. 

Using the metrics part of OpenTelemetry.Instrumentation.Owin is currently an impossibility due to this.

### PR DESCRIPTION
**Pull request by gliljas**
Fixes #1654.

## Changes

Made both the Meter and Histogram singletons

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [x] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.Owin/Implementation/OwinInstrumentationMetrics.cs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // Copyright The OpenTelemetry Authors
 // SPDX-License-Identifier: Apache-2.0
 
+using System;
```
Comment (Kielek) [inline src/OpenTelemetry.Instrumentation.Owin/Implementation/OwinInstrumentationMetrics.cs:4]:
```suggestion
```

**Review by Kielek — COMMENTED** [review]
Please add entry to the CHANGELOG.md file. It is important bugfix, we should document it properly to inform users about importance of update.

**Comment by gliljas** [comment]
> Please add entry to the CHANGELOG.md file. It is important bugfix, we should document it properly to inform users about importance of update.

Entry added with the word "massive" added for urgency. I think a new package should be released as soon as possible. RC4 is a server killer.

**Review by Kielek — APPROVED** [review]

**Comment by Kielek** [comment]
@gliljas, could you please create request release PR similar to https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1645/files ?

Then I can make a release.

**Comment by Kielek** [comment]
@gliljas, please check https://www.nuget.org/packages/OpenTelemetry.Instrumentation.Owin/1.0.0-rc.6

**Pull request merged**